pkg/service/api_service_test.go

```diff
@@ -122,28 +122,76 @@ var (
 			},
 		},
 	}
+	// A secret that seems to have the correct annotation and data, but is not of type service account token
+	wrongSecret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "bootstrap-token",
+			Namespace: clusterA.Namespace,
+			Annotations: map[string]string{
+				"kubernetes.io/service-account.name": clusterA.Name,
```

Comment:
I don't mind either way for the tests, but I'd probably use `corev1.ServiceAccountNameKey` here as well.